code/modules/mob/living/carbon/human/species_types/shadowpeople.dm

```diff
@@ -333,9 +377,290 @@
+
+
+/datum/movespeed_modifier/shadow_sect
+	multiplicative_slowdown = -0.75
+
+
+/obj/item/organ/heart/shadow_ritual1
+	name = "shadowed heart"
+	desc = "This heart seems to be covered in shadows, even under the strong light."
+	icon = 'icons/obj/surgery.dmi'
+	icon_state = "shadow_heart_1"
+	visual = TRUE
+	decay_factor = 0
+	var/shadow_conversion = 0
+	var/datum/action/innate/shadow_coms/comm/coms = new
+
+/obj/item/organ/heart/shadow_ritual2
+	name = "faded heart"
+	desc = "It's hard to make out a heart under the cover of this ever-shifting darkness."
+	icon = 'icons/obj/surgery.dmi'
+	icon_state = "shadow_heart_2"
+	visual = TRUE
+	decay_factor = 0
+	var/shadow_conversion = 0
+	var/datum/action/innate/shadow_coms/comm/coms = new
+
+/obj/item/organ/heart/shadow_ritual3
+	name = "pulsing darkness"
+	desc = "You can't see the object covered in darkness, no light can dispel it. You can only see how the darkness itself moves, pulsing, time and time again."
+	icon = 'icons/obj/surgery.dmi'
+	icon_state = "shadow_heart_3"
+	visual = TRUE
+	decay_factor = 0
+	var/respawn_progress = 0
+	var/shadow_conversion = 0
+	var/datum/action/innate/shadow_coms/comm/coms = new
+
+
+/obj/item/organ/heart/shadow_ritual1/Stop()
+	return 0
+
+/obj/item/organ/heart/shadow_ritual2/Stop()
+	return 0
+
+/obj/item/organ/heart/shadow_ritual3/Stop()
+	return 0
+
+
+/obj/item/organ/heart/shadow_ritual1/update_icon()
+	return
+
+/obj/item/organ/heart/shadow_ritual2/update_icon()
+	return
+
+/obj/item/organ/heart/shadow_ritual3/update_icon()
+	return
+
+
+/obj/item/organ/heart/shadow_ritual1/Insert(mob/living/carbon/M, special = 0, pref_load = FALSE)
+	..()
+	if(isshadow(M))
+		var/mob/living/carbon/human/S = M
+		var/datum/species/shadow/spiec = S.dna.species
+		spiec.shadow_sect_dependency = 1
+		coms.Grant(M)
+	else
+		shadow_conversion = 0
+		to_chat(M, span_userdanger("You feel a cold feeling spreading from your chest. This might not have been a great idea."))
+
+/obj/item/organ/heart/shadow_ritual2/Insert(mob/living/carbon/M, special = 0, pref_load = FALSE)
+	..()
+	if(isshadow(M))
+		var/mob/living/carbon/human/S = M
+		var/datum/species/shadow/spiec = S.dna.species
+		spiec.shadow_sect_dependency = 2
+		coms.Grant(M)
+	else
+		shadow_conversion = 0
+		to_chat(M, span_userdanger("You feel a cold feeling spreading from your chest. This might not have been a great idea."))
+
+/obj/item/organ/heart/shadow_ritual3/Insert(mob/living/carbon/M, special = 0, pref_load = FALSE)
+	..()
+	if(isshadow(M))
+		var/mob/living/carbon/human/S = M
+		var/datum/species/shadow/spiec = S.dna.species
+		spiec.shadow_sect_dependency = 3
+		coms.Grant(M)
+	else
+		shadow_conversion = 0
+		to_chat(M, span_userdanger("You feel a cold feeling spreading from your chest. This might not have been a great idea."))
+
+
+/obj/item/organ/heart/shadow_ritual1/Remove(mob/living/carbon/M, special = 0, pref_load = FALSE)
```

Comment:
All of these are doing the exact same thing. Create a parent type with the logic, and declare the 1,2,3 items by subtype only.

```
/datum/animal/proc/do_something()
  // Do some shared logic

/datum/animal/dog
  name = "Dog"

/datum/animal/cat
  name = "Cat"
```